lib/active_record/connection_adapters/trilogy_adapter.rb

```diff
@@ -153,6 +153,91 @@ def discard!
         self.connection = nil
       end
 
+      # ActiveRecord 7.0 support
+      if ActiveRecord.version < ::Gem::Version.new('7.1.a')
+        def initialize(connection, logger, connection_options, config)
+          super
+          if @connection
+            @verified = true
+            @raw_connection = @connection
+          end
+          # Ensure that we're treating prepared_statements in the same way that Rails 7.1 does
+          @prepared_statements = self.class.type_cast_config_to_boolean(
+            @config.fetch(:prepared_statements) { default_prepared_statements }
+          )
+        end
+
+        def connect!
+          verify!
+          self
+        end
+
+        def reconnect!
+          @lock.synchronize do
+            disconnect!
+            connect
+          rescue StandardError => original_exception
+            @verified = false
+            raise translate_exception_class(original_exception, nil, nil)
+          end
+        end
+
+        def exec_rollback_db_transaction
+          # 16384 tests the bit flag for SERVER_SESSION_STATE_CHANGED, which gets set when the
+          # last statement executed has caused a change in the server's state.
+          if active? || (@raw_connection.server_status & 16_384).positive?
+            super
+          else
+            @verified = false
+          end
+        end
+
+        def with_raw_connection(uses_transaction: true, **_kwargs)
```

Comment:
I'm particularly concerned that other gems might feature-detect for this method and then assume they're talking to AR 7.1+. Perhaps we call this `with_trilogy_connection`, then alias it in the new-Rails case?

Reply:
Sounds good -- ended up only needing to skip 3 examples:
* #execute answers results for valid query after losing connection unexpectedly
* #execute fails if the connection is closed
* can reconnect after failing to commit
Perhaps at a minimum these are the specific ones that can now be removed altogether.

Reply:
Great idea -- have renamed this in [6234970](https://github.com/github/activerecord-trilogy-adapter/pull/26/commits/62349704862c856e73e7f12f6228b75eb7e9e94d).